src/anemoi/graphs/inspector.py

```diff
@@ -0,0 +1,200 @@
+        """
+        edge_summary = []
+        for (src_name, _, dst_name), edges in self.graph.edge_items():
+            attributes = edges.edge_attrs()
+            attributes.remove("edge_index")
+
+            edge_summary.append(
+                [
+                    src_name,
+                    dst_name,
+                    number(edges.num_edges),
+                    number(self.graph[src_name].num_nodes - len(torch.unique(edges.edge_index[0]))),
+                    number(self.graph[dst_name].num_nodes - len(torch.unique(edges.edge_index[1]))),
+                    sum(edges[attr].shape[1] for attr in attributes),
+                    ", ".join(attributes),
+                ]
+            )
+        return edge_summary
+
+    def describe(self) -> None:
+        """Describe the graph."""
+        print()
+        print(f"📦 Path       : {self.path}")
+        print(f"💽 Size       : {bytes(self.total_size)} ({number(self.total_size)})")
+        print()
+        print(
+            table(
+                self.get_node_summary(),
+                header=[
+                    "Nodes name",
+                    "Num. nodes",
+                    "Attributes",
+                    "Attribute dim",
+                    "Min. latitude",
+                    "Max. latitude",
+                    "Min. longitude",
+                    "Max. longitude",
+                ],
+                align=["<", ">", ">", ">", ">", ">", ">", ">"],
+                margin=3,
+            )
+        )
+        print()
+        print()
+        print(
+            table(
+                self.get_edge_summary(),
+                header=[
+                    "Source",
+                    "Target",
+                    "Num. edges",
+                    "Isolated Source",
+                    "Isolated Target",
+                    "Attribute dim",
+                    "Attributes",
+                ],
+                align=["<", "<", ">", ">", ">", ">", ">"],
+                margin=3,
+            )
+        )
+        print("🔋 Graph ready.")
+        print()
+
+
+class GraphInspectorTool:
+    """Inspect the graph."""
+
+    def __init__(
+        self,
+        path: Union[str, Path],
+        output_path: Path,
+        show_attribute_distributions: Optional[bool] = True,
+        show_nodes: Optional[bool] = False,
+        **kwargs,
+    ):
+        self.path = path
+        self.graph = torch.load(self.path)
+        self.output_path = output_path
+        self.show_attribute_distributions = show_attribute_distributions
+        self.show_nodes = show_nodes
+
+        if isinstance(self.output_path, str):
+            self.output_path = Path(self.output_path)
+
+        os.makedirs(self.output_path, exist_ok=True)
+
+        assert self.output_path.is_dir(), f"Path {self.output_path} is not a directory."
+        assert os.access(self.output_path, os.W_OK), f"Path {self.output_path} is not writable."
+
+    def run_all(self):
```

Comment:
I would propose to call this method 
`def inspect(self)`
to be more consistent with the description tool